### PR TITLE
Fix task property annotation problems

### PR DESCRIPTION
**Pull request by lptr**
This PR fixes some validation problems with tasks in the plugin that will cause any build using the plugin to fail with Gradle 7.0. It also enables strict validation during the build, so any new validation problems introduced will make the build fail.

The particular problems fixed by this PR are:

```text
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':jruby-gradle-core-plugin:validateTaskProperties'.
> Task property validation failed. See https://docs.gradle.org/5.4.1/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Task type 'com.github.jrubygradle.api.core.AbstractJRubyPrepare': property 'projectOperations' is not annotated with an input or output annotation.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':jruby-gradle-plugin:validateTaskProperties'.
> Task property validation failed. See https://docs.gradle.org/5.4.1/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Task type 'com.github.jrubygradle.GenerateGradleRb': property 'destinationDir' is not annotated with an input or output annotation.
   > Warning: Task type 'com.github.jrubygradle.GenerateGradleRb': property 'gemInstallDir' is not annotated with an input or output annotation.
   > Warning: Task type 'com.github.jrubygradle.JRubyExec': property 'gemWorkDir' is not annotated with an input or output annotation.
   > Warning: Task type 'com.github.jrubygradle.JRubyExec': property 'jrubyVersion' is not annotated with an input or output annotation.
   > Warning: Task type 'com.github.jrubygradle.JRubyExec': property 'script' has @Input annotation used on property of type java.io.File.
   > Warning: Task type 'com.github.jrubygradle.JRubyPrepare': property 'projectOperations' is not annotated with an input or output annotation.
```

### CODE REVIEW

**Review by melix — APPROVED** [review]

**Comment by lptr** [comment]
Please also note that these warnings need to be fixed in order to fix the warnings in Asciidoctor: https://github.com/asciidoctor/asciidoctor-gradle-plugin/pull/587

Without these changes it is not possible to produce a version of the Asciidoctor plugins that are compatible with Gradle 7.0.

**Review by rtyler — APPROVED** [review]
Overall this looks good to me, thanks for the pull request! I'd like @ysb33r to take a gander before I merge it since he's much more plugged into the modern day Gradle than I am

**Comment by lptr** [comment]
Thanks for the review, @rtyler. Given that we need the changes here to fix Asciidoctor, is there a way to do a quick release of this plugin once the PR is merged?

**Comment by rtyler** [comment]
@lptr there certainly is, but also [ffs](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/)

**Comment by ysb33r** [comment]
@lptr Be aware that this will update an alpha version of this plugin which incidentally also addresses configuration cache. Asciidoctor Gradle 3.3.0 still uses the non-alpha version. If you need this fix to address Asciidoctor 3.3.x, then we'll need to make a PR against a branch of jrunbtgradle.

**Comment by lptr** [comment]
@ysb33r thanks for merging. Yes, we'd need this to fix Asciidoctor Gradle plugin 3.3.x. What is jrunbtgradle? How can I help make the changes so that we can unblock Asciidoctor's Gradle 7.0 support?

**Pull request merged**
